### ISSUE TITLE
Add and evaluate reco methods

### ISSUE THREAD

**Issue by heidemeissner** [Ptychography-4-0/ptychography#11]
First step: add reconstruction methods from WP2 plus interfaces to libertem
Second step: Evaluate concerning:
   - applicability and limitations of the algorithms
   - performance analysis (parallelization, scalability)
   - accuracy (e.g. robustness to noise)
   - convergence to correct / unique solution (expected)
   - applicability to interactive measurements ("online reconstruction")?
   - data formats and data handling, e.g. zeroMQ, Check with WP1 colleagues

**Comment by heidemeissner** [comment]
Issue: uncertainties in raster point positions and positions of diffraction images due to shifts of beam and objects. Which algorithm is most robust?

**Comment by heidemeissner** [comment]
Check for further solvers of optimization problems, e.g. in Pytorch and tensorFlow. First step: define cost function

**Comment by heidemeissner** [comment]
Considered algorithms are:
- EPIE (DESY)
- EPIE (Achim)
- direct solver (phase lift)
- Kurato-Vicsek?
- PtyPy?
- Difference Map?
- Wirtinger Flow?
- (Blind) Wigner Deconvolution
